doc/nrf/libraries/networking/nrf_provisioning.rst

```diff
@@ -0,0 +1,162 @@
+* :kconfig:option:`CONFIG_NRF_PROVISIONING_HTTP_RX_BUF_SZ` - HTTP response payload buffer size
+* :kconfig:option:`CONFIG_NRF_PROVISIONING_HTTP_TX_BUF_SZ` - HTTP request body size
+* :kconfig:option:`CONFIG_NRF_PROVISIONING_HTTP_JWT` - Chooses JWT token for client authentication
+* :kconfig:option:`CONFIG_NRF_PROVISIONING_HTTP_ATTESTTOKEN` - Chooses attestation token for client authentication
+
+.. _lib_nrf_provisioning_start:
+
+Usage
+*****
+
+The usage of the nRF Device provisioning library is described in the following sections.
+
+Initialization
+==============
+
+Once initialized, the provisioning client will run on its own in the background.
+The provisioning client can be initialized in following two ways:
+
+* The application calls :c:func:`nrf_provisioning_init` which starts the client.
+* Set the client to initialize during Zephyr's system initialization phase.
+  In this case it is assumed that a network connection has been established in the same phase.
+
+The function uses the following two arguments:
+
+*  A pointer to a callback function which is to be called on modem state changes.
+*  A pointer that must be called once provisioning is done.
+
+If you provide ``null`` as a callback function address argument, a corresponding default callback used.
+Consequent calls to the initialization function will only change the callback functions.
+This behavior is beneficial when the client has been initialized during the system initialization phase, but the application wants to register its own callback functions afterwards.
+
+Provisioning
+============
+
+By default, when provisioning is done after receiving the ``FINISHED`` command, the device is rebooted.
+The behavior can be overwritten by providing a unique callback function for the initialization function.
+
+If anything is written on the modem's non-volatile memory, the modem needs to be set in offline mode.
+This is because the modem cannot be connected when any data is to be written on to its storage area.
+After the memory write is completed, the aforementioned callback function must be called again to set the modem in desired state.
+To use the default implementation, ``NULL`` can be passed as an argument to the :c:func:`nrf_provisioning_init` function.
+The default callback function must be copied and modified based on your needs if it is necessary.
+
+The library starts provisioning when it initializes and afterwards it is done in a configured interval.
+The interval is read from the storage settings, and it can be updated with a provisioning command like any other key-value pair.
+
+During provisioning, the library first tries to establish the transport for communicating with the service.
+This procedure involves a TLS handshake where the client establishes the correct server.
+The server uses the attestation token or the JWT token generated by the device for authenticating the client.
```

Comment:
Cloud-side docs on attestation tokens and the Identity Service will be available next week.